XenAdmin/Diagnostics/Problems/HostProblem/HostOutOfSpaceProblem.cs

```diff
@@ -115,5 +115,11 @@ public override bool IsFixable
                 return false;
             }
         }
+
+        public override bool Equals(object obj)
+        {
+            var problem = obj as HostOutOfSpaceProblem;
+            return problem != null && diskSpaceReq.Equals(problem.diskSpaceReq);
```

Comment:
I am afraid there is a chance for getting a null reference exception. (diskSpaceReq)